tools/Environments/DevHome.Environments/ViewModels/ComputeSystemViewModel.cs

```diff
@@ -348,17 +359,26 @@ public void Receive(ComputeSystemOperationCompletedMessage message)
             _log.Information($"operation '{data.ComputeSystemOperation}' completed for Compute System: {Name}");
 
             var completionStatus = EnvironmentsTelemetryStatus.Succeeded;
+            var providerId = ComputeSystem.AssociatedProviderId.Value;
 
             if ((data.OperationResult == null) || (data.OperationResult.Result.Status == ProviderOperationStatus.Failure))

```

Comment:
These if checks and the resulting event generation are copied in multiple places.  Is it possible to eventually move the code to a common location?

Reply:
I consolidated them into a single helper function in the ComputeSystemHelpers.cs file. Thanks